packages/core/src/components/advanced-select/advanced-select.tsx

```diff
@@ -0,0 +1,160 @@
+import { Component, ComponentInterface, Prop, h, FunctionalComponent, Event, EventEmitter } from "@stencil/core";
+
+import {
+  AdvancedSelectGroupRedirect,
+  AdvancedSelectOption,
+  AdvancedSelectOptionsOrGroup,
+} from "./advanced-select.models";
+import {
+  AdvancedSelectClickEvent,
+  AdvancedSelectOptionClickEvent,
+  AdvancedSelectRedirectClickEvent,
+} from "./advanced-select.interfaces";
+
+@Component({
+  tag: "dso-advanced-select",
+  styleUrl: "advanced-select.scss",
+  shadow: true,
+})
+export class AdvancedSelect implements ComponentInterface {
+  /**
+   * The options to display in the select.
+   */
+  @Prop()
+  options: AdvancedSelectOptionsOrGroup<any>[] = [];
+
+  /**
+   * The active option. By object reference.
+   */
+  @Prop()
+  active?: AdvancedSelectOption<any>;
+
+  /**
+   * An extra text for the active option. Only visible in the list of options.
+   */
+  @Prop()
+  activeHint?: string;
+
+  /**
+   * The open state of the options list.
+   */
+  @Prop()
+  open: boolean = false;
+
+  /**
+   * Emitted when user clicks the select.
+   */
+  @Event({ bubbles: false })
+  dsoClick!: EventEmitter<AdvancedSelectClickEvent>;
+
+  /**
+   * Emitted when user clicks an option
+   */
+  @Event({ bubbles: false })
+  dsoOptionClick!: EventEmitter<AdvancedSelectOptionClickEvent<any>>;
+
+  /**
+   * Emitted when user clicks a redirect link.
+   */
+  @Event({ bubbles: false })
+  dsoRedirectClick!: EventEmitter<AdvancedSelectRedirectClickEvent>;
+
+  private handleClick = (event: MouseEvent) => {
+    this.dsoClick.emit({ originalEvent: event });
+  };
+
+  private handleOptionClick = (event: MouseEvent, option: AdvancedSelectOption<any>) => {
+    this.dsoOptionClick.emit({ originalEvent: event, option });
+  };
+
+  private handleRedirectClick = (event: MouseEvent, redirect: AdvancedSelectGroupRedirect) => {
+    this.dsoRedirectClick.emit({ originalEvent: event, redirect });
+  };
+
+  render() {
+    return (
+      <div>
+        <button class={{ "active-option": true, open: this.open }} type="button" onClick={this.handleClick}>
+          <span class="active-option-label">{this.active?.label ?? "Selecteer een optie"}</span>
+          <span class="active-option-aside">
+            {this.options.some((option) => "summaryCounter" in option && option?.summaryCounter) ? (
+              <span class="badges">
+                {this.options
+                  .filter((option) => "summaryCounter" in option && option?.summaryCounter)
```

Comment:
Volgens mij houd je nu geen rekening met een Option met `summaryCounter: true` in een Group.

Reply:
Hoe bedoel je dit precies? `summaryCounter` bestaat alleen in `AdvancedSelectGroup`.